host/tests/gatt.rs

```diff
@@ -41,20 +41,20 @@ async fn gatt_client_server() {
         // Random starting value to 'prove' the incremented value is correct
         let mut value: [u8; 1] = [rand::prelude::random(); 1];
         let mut expected = value[0].wrapping_add(1);
-        let mut svc = table.add_service(Service::new(0x1800));
-        let _ = svc.add_characteristic_ro(0x2a00, id);
-        let _ = svc.add_characteristic_ro(0x2a01, &appearance[..]);
+        let mut svc = table.add_service(Service::new(0x1800), None);
+        let _ = svc.add_characteristic_ro(0x2a00, id, None);
+        let _ = svc.add_characteristic_ro(0x2a01, &appearance, None);
         svc.build();
 
         // Generic attribute service (mandatory)
-        table.add_service(Service::new(0x1801));
+        table.add_service(Service::new(0x1801), None);
 
         // Custom service
-        let value_handle = table.add_service(Service::new(SERVICE_UUID.clone()))
+        let value_handle = table.add_service(Service::new(SERVICE_UUID.clone()), None)
             .add_characteristic(
                 VALUE_UUID.clone(),
                 &[CharacteristicProp::Read, CharacteristicProp::Write, CharacteristicProp::Notify],
-                &mut value)
+                &mut value, None, None)
```

Comment:
It's a bit annoying that these optional parameters creep up in the API. I think it would be better to use some kind of builder pattern to avoid that.